File: PORTAS.c
#include <stdio.h>
int processamento(int a, int b, int op){
	switch(op){
		case 1: //NOT
			return !a;
		case 2: //NAND
			return !(!(a&b));
		case 3: //NORna
			return !(!(a|b));
		case 4: //AND
			return !(a&b);
		case 5: //OR
			return !(a|b);
		default:
			return -1;
	}	
}

int main(){
	//testando na porta OR com valores A  e B
    //basta trocar os valores que corresponde a tabela
	printf ("!(%d or %d) = %d\n",1,0,processamento(1,0,5));
                              // a,b             //a,b,op
	return 0;
}

File: Portas.rs


fn main() {
    //testando na porta OR com valores A  e B
    //basta trocar os valores que corresponde a tabela
    println!("!({} or {}) = {}\n",0,1,processamento(0,1,5)); //parametros do processamento
                               // a,b             //a,b,op
}

fn processamento(a:i32,b:i32,op: i32) -> i32{
  
    if op == 1{//PORTA NOT
        !(!a)
    }else if op == 2{ //PORTA NAND 
        !(!(a&b))
    }else if op == 3{//PORTA NOR
        !(!(a|b))
    }else if op == 4{//PORTA AND
        !(a&b)+2
    }else if op == 5{//PORTA OR
        !(a|b)+2
    }else{
        -1
    }
}
